docs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,196 @@
+# Troubleshooting
+
+## Terminology
+
+See [GLOSSARY.md](./GLOSSARY.md).
+
+- - -
+
+## Problems
+
+- [Common issues](#common-issues)
+- [Caller does not have permission in the Organization](#Caller-does-not-have-permission-in-the-organization)
+- [Billing quota exceeded](#Billing-quota-exceeded)
+
+- - -
+
+### Common issues
+
+- [Project quota exceeded](#project-quota-exceeded)
+- [Terraform State Snapshot lock](#terraform-state-snapshot-lock)
+- [Application authenticated using end user credentials](#application-authenticated-using-end-user-credentials)
+- [Cannot assign requested address error in Cloud Shell](#cannot-assign-requested-address-error-in-cloud-shell)
+
+#### Project quota exceeded
+
+**Error message:**
+
+```
+Error code 8, message: The project cannot be created because you have exceeded your allotted project quota
+```
+
+**Cause:**
+
+This message means you have reached your [project creation quota](https://support.google.com/cloud/answer/6330231).
+
+**Solution:**
+
+In this case, you can use the [Request Project Quota Increase](https://support.google.com/code/contact/project_quota_increase)
+form to request a quota increase.
+
+In the support form,
+for the field **Email addresses that will be used to create projects**,
+use the email address of `terraform_service_account` that is created by the Terraform Example Foundation 0-bootstrap step.
+
+**Notes:**
+
+- If you see other quota errors, see the [Quota documentation](https://cloud.google.com/docs/quota).
+
+#### Terraform State Snapshot lock
+
+**Error message:**
+
+When running the build for the branch `production` in step 3-networks in your **Foundation Pipeline** the build fails with:
+
+```
+state snapshot was created by Terraform v0.x.x, which is newer than current v0.13.6; upgrade to Terraform v0.x.x or greater to work with this state
+```
+
+**Cause:**
+
+The manual deploy step for the shared environment in [3-networks](../3-networks#deploying-with-cloud-build) was execute with a Terraform version newer than version v0.13.6 used in the **Foundation Pipeline**.
+
+**Solution:**
+
+You will need to re-run the deploy of the 3-networks shared environment with Terraform v0.13.6
```

Comment:
I think we should also specify the alternative where they edit the bootstrap and create a 0.x.x terraform runner image. We expect foundation to be mostly compatible especially if 0.x.x worked fine when running locally.

Reply:
Done.
Instructions tested in one of my deploys.